src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldServer.java

```diff
@@ -75,4 +84,37 @@ public void onPostInit(CallbackInfoReturnable<World> ci) {
         IMixinWorld world = (IMixinWorld) ci.getReturnValue();
         world.updateWorldGenerator();
     }
+
+    @Overwrite
+    public void updateAllPlayersSleepingFlag() {
+        this.allPlayersSleeping = false;
+        if (!this.playerEntities.isEmpty()) {
+            int ignoredPlayers = 0;
+            int sleepingPlayers = 0;
+            for (EntityPlayer entityPlayer : this.playerEntities) {
+                if (entityPlayer.isSpectator()
+                        || (entityPlayer instanceof Human && ((Human)entityPlayer).isSleepingIgnored())) {
```

Comment:
Humans are no longer types of `EntityPlayer`. They are their own entity type, [`EntityHuman`](https://github.com/SpongePowered/SpongeCommon/blob/master/src/main/java/org/spongepowered/common/entity/living/human/EntityHuman.java#L74) which extends `EntityCreature`. But as I said in your API PR the sleep flag is only for players.